hack/machineset.sh

```diff
@@ -287,7 +287,7 @@ EOF
 }
 
 # Retrieves the Cloud Provider for the OpenShift Cluster
-provider="$(oc -n openshift-kube-apiserver get configmap config -o json | jq -r '.data."config.yaml"' | jq '.apiServerArguments."cloud-provider"' | jq -r '.[]')"
+platform="$(oc get infrastructure cluster -ojsonpath={.spec.platformSpec.type})"
```

Comment:
Will this work for older clusters?

Reply:
It should, its the same method we use in the e2e tests: https://github.com/openshift/windows-machine-config-operator/blob/master/test/e2e/providers/cloudprovider.go#L34